### PR TITLE
drivers: tlv320dac310x: Add DTS support for codec reset GPIO

### PR DESCRIPTION
**Pull request by sathishkuttan**
This PR adds support for a DTS defined GPIO to be used as a reset signal for codec reset.
The codec reset GPIO is defined in DTS, the driver binds to the GPIO driver instance and uses the GPIO driver to control the reset signal.
See also #11636

### CODE REVIEW

**Review by SavinayDharmappa — APPROVED** [review]

File: drivers/audio/tlv320dac310x.c [hunk 1]
```diff
@@ -33,8 +38,9 @@ struct codec_driver_data {
 
 static struct codec_driver_config codec_device_config = {
 	.i2c_device	= NULL,
-	.i2c_dev_name	= DT_CODEC_I2C_BUS_NAME,
```
Comment (galak) [inline drivers/audio/tlv320dac310x.c]: the change to use DT_CODEC_I2C_BUS_NAME instead of .i2c_dev_name/.i2c_address should NOT be part of this commit/PR.  Please remove them.

Its actually better to leave the code as is, so its closer to supporting multiple instances of this i2c device.

**Review by galak — CHANGES_REQUESTED** [review]
Please drop the i2c name/addr changes from this PR.

File: drivers/audio/tlv320dac310x.c (hunk 1 above)
Comment (sathishkuttan) [reply drivers/audio/tlv320dac310x.c]: Good catch. I have retained the I2C name/address in the structure and also added the GPIO pin/flags.

**Review by galak — APPROVED** [review]
